deltachat-ffi/deltachat.h

```diff
@@ -4882,7 +4910,44 @@ uint32_t        dc_lot_get_id            (const dc_lot_t* lot);
  * @param lot The lot object.
  * @return The timestamp as defined by the creator of the object. 0 if there is not timestamp or on errors.
  */
-int64_t          dc_lot_get_timestamp     (const dc_lot_t* lot);
+int64_t         dc_lot_get_timestamp     (const dc_lot_t* lot);
+
+
+/**
+ * Returns array of contacts which reacted to the given message.
+ *
+ * @memberof dc_reactions_t
+ * @param reactions The object containing message reactions.
+ * @return array of contact IDs. Use dc_array_get_cnt() to get array length and
+ *         dc_array_get_id() to get the IDs.
+ */
+dc_array_t*     dc_reactions_get_contacts(dc_reactions_t* reactions);
```

Comment:
some thoughts of the `dc_reactions_t` API, that also others may have during review:

UIs of at least whatsapp/signal show the reaction-details by emojis as tab atop, each tap show the contacts:

<img width=320 src=https://user-images.githubusercontent.com/9800740/196285150-6e8572bf-7665-426b-b204-717a5f906538.jpg>

the reaction-summary in the bubble is rendered similar, just without showing each contact.

so, from the view of the UI, _at a first glance,_ having the API the other way round is more practical, so a `dc_reactions_get()` returning an array of reaction-strings (for the tabs and bubbles) and then a `dc_reactions_get_contacts()` taking one reaction-strings and return the contacts (for the numbers in the bubbles and for the content of each tab).

also, that way, the splitting by string would be left up to the core and we could also define the emoji order by the core as we like.

i just want to bring up these points, maybe at the end each UI will have anyway a high level single `getReactions()` that returns a complex datatype.

so, also _at a first glance_, seems the approach currently implemented in the pr is good enough as well.

Reply:
yip, i think it is fine as it is. and yes, let's add high-level getters as needed in subsequent prs.